pages/pipelines/plugins.md.erb

```diff
@@ -3,73 +3,74 @@
 
 Plugins provide a way to extend the functionality that Buildkite natively offers. They act as a external repository of [Hooks][hooks] that are referenced from your `pipeline.yml` file, downloaded and run as part of your builds.
 
-Plugins can be hosted on GitHub, BitBucket or a private git repository, and are automatically fetched as required.
+Plugins can be hosted on GitHub, BitBucket or a private git repository, and are automatically fetched as required. See [Plugin sources] for how you can host your own.
 
 Buildkite offers a [collection of useful plugins][Buildkite Plugins]
-to get you started, including a [Docker Compose plugin] and a [Junit Annotate plugin].
+to get you started, including a [Docker Compose plugin], [Junit Annotate plugin], and the [Library Example plugin].
 
 <%= toc %>
 
 ## Using a plugin
 
-Plugins are only available for [pipelines] using `pipeline.yml` files.
+Plugins can only be used in [command steps] in `pipeline.yml` [pipeline definitions].
 
-Plugins can be used in two ways:
-
-* as a standalone step, performing an operation
-* as a library, making commands available to that step's script
-
-An example of what a plugin might look like in your code:
+The simplest plugin is one that accepts no configuration, such as the [Detect Clowns plugin], which will return an error if any clown emojis are found in your source code:
 
 ```yml
 steps:
-  - plugins:
-      detect-clowns#v2.0.0: ~
+  - label: "\:clown_face\:"
+    plugins:
+      detect-clowns#v1.0.0: ~
 ```
 
-This example will install the `buildkite-plugins/detect-clowns-buildkite-plugin` when your pipeline runs this build step.
+More commonly, plugins accept various configuration options, as the [Docker plugin] which can be configured to run a build step using any publicly available Docker image:
 
-Plugins are automatically fetched from where they are hosted, based on the name of the project. See [Plugin sources] for how you can host your own.
+```yml
+steps:
+  - command: yarn install && yarn run test
+    plugins:
+      docker#v1.0.0:
+        image: node
+        workdir: /app
+```
 
-A real-life example is the [Docker Compose plugin]. It provides an integration
-with docker-compose that assists with running your builds within a docker environment:
+More advanced plugins, such as [Docker Compose plugin], can be run multiple times within a build, and use [buildkite-agent meta-data] to share data between the steps:
```

Comment:
What does 'can be run multiple times in a build' mean? I think we need to explain a bit earlier (before the docker plugin example too), what it means to 'run' a plugin. In the first line we say that the hooks are 'downloaded and run as part of your builds', maybe that's the point where we could be clearer? 

I don't think this comment is very clear either 😅I'm mostly confused by the bits that say that the plugins are hooks, but they're 'run' like a script, but also that there's configuration too and I'm not sure where that comes in/how it's executed/what it's for?

Reply:
Thanks, they're good questions! I've tried to clarify how that works in the latest commit…